src/MonteCarlo/MonteCarlos.jl

```diff
@@ -73,16 +73,26 @@ Expression tree optimization using Monte Carlo with parameters p, grammar 'gramm
 function monte_carlo(p::MonteCarlo, grammar::Grammar, typ::Symbol, loss::Function; 
     verbose::Bool=false)
     dmap = mindepth_map(grammar)
-    best_tree, best_loss = RuleNode(0), Inf
```

Comment:
The original implementation is memory-efficient and friendly for `_update_tracker!`, which cannot be assumed to be thread-safe.